src/Microsoft.AspNetCore.Blazor/Components/BlazorComponent.cs

```diff
@@ -84,6 +84,19 @@ protected virtual void OnParametersSet()
         protected virtual Task OnParametersSetAsync()
             => null;
 
+        /// <summary>
+        /// Method invoked when the component has rendered into the DOM.
+        /// </summary>
+        protected virtual void OnRenderComplete()
+        {
+        }
+
+        /// <summary>
+        /// Method invoked when the component has rendered into the DOM.
+        /// </summary>
+        protected virtual Task OnRenderCompleteAsync()
+            => null;
```

Comment:
Wouldn't be best to return `Task.CompletedTask` instead of `null`? That avoid unnecessary null check(s) and don't allocate an extra `Task` either.